core/src/main/scala/codesearch/core/search/Search.scala

```diff
@@ -73,12 +91,16 @@ trait Search {
       case None           => if (request.sourcesOnly) extensionsRegex else ".*"
     }
 
-    val query: String =
-      RegexConstructor(request.query, request.insensitive, request.spaceInsensitive, request.preciseMatch)
+    val query: String = {
+      val preciseMatch: String = if (request.preciseMatch) PreciseMatch(request.query) else request.query
+      if (request.spaceInsensitive) SpaceInsensitiveString(preciseMatch) else preciseMatch
+    }
 
     request.filter match {
-      case Some(filter) => List("csearch", "-n", "-f", forExtensions, query, filter)
-      case None         => List("csearch", "-n", "-f", forExtensions, query)
+      case Some(filter) if request.insensitive => List("csearch", "-n", "-i", "-f", forExtensions, query, filter)
```

Comment:
It is not correct - insensitive flag adding to query via RegexConstructor. Check actual develop branch.